app/src/main/java/tech/ula/MainActivity.kt

```diff
@@ -161,6 +170,65 @@ class MainActivity : AppCompatActivity(), SessionListFragment.SessionSelection,
         ACRA.init(applicationContext as Application, builder)
     }
 
+    private fun setupReviewRequest() {
+        val minimumTimePassedReached = usageUtility.getIsSufficientTimeElapsedSinceFirstOpen()
+        val minimumTimesOpenedReached = usageUtility.numberOfTimesOpenedIsGreaterThanThreshold()
+        val userAlreadyGaveReviewOrDismissed = usageUtility.getUserGaveFeedback()
+
+        if (minimumTimePassedReached && minimumTimesOpenedReached && !userAlreadyGaveReviewOrDismissed) {
+            val requestReviewView = layoutInflater.inflate(R.layout.list_item_review_request, null)
+            requestReviewView.visibility = View.VISIBLE
+            val viewHolder = findViewById<ViewGroup>(R.id.request_review_insert_point)
+            viewHolder.addView(requestReviewView, 0)
+            setupReviewRequestUI(viewHolder)
+        }
+    }
+
+    private fun setupReviewRequestUI(viewHolder: ViewGroup) {
+        val requestQuestion = viewHolder.findViewById<TextView>(R.id.prompt_review_question)
+        val negativeBtn = viewHolder.findViewById<Button>(R.id.btn_negative_response)
+        val positiveBtn = viewHolder.findViewById<Button>(R.id.btn_positive_response)
+
+        positiveBtn.setOnClickListener {
+            requestQuestion.text = getString(R.string.review_ask_for_rating)
+            positiveBtn.text = getString(R.string.button_positive)
+            negativeBtn.text = getString(R.string.button_negative_1)
+
+            positiveBtn.setOnClickListener {
+                userGaveReviewOrDismissed(viewHolder)
+                val userlandPlayStoreURI = "https://play.google.com/store/apps/details?id=tech.ula"
+                val intent = Intent("android.intent.action.VIEW", Uri.parse(userlandPlayStoreURI))
+                startActivity(intent)
+            }
+
+            negativeBtn.setOnClickListener {
+                userGaveReviewOrDismissed(viewHolder)
+            }
+        }
+
+        negativeBtn.setOnClickListener {
+            requestQuestion.text = getString(R.string.review_ask_for_feedback)
+            positiveBtn.text = getString(R.string.button_positive)
+            negativeBtn.text = getString(R.string.button_negative_2)
+
+            positiveBtn.setOnClickListener {
+                userGaveReviewOrDismissed(viewHolder)
+                val githubURI = "https://github.com/CypherpunkArmory/UserLAnd"
+                val intent = Intent("android.intent.action.VIEW", Uri.parse(githubURI))
+                startActivity(intent)
+            }
+
+            negativeBtn.setOnClickListener {
+                userGaveReviewOrDismissed(viewHolder)
+            }
+        }
+    }
+
+    private fun userGaveReviewOrDismissed(viewHolder: ViewGroup) {
```

Comment:
I think this function could more appropriately be named `handleUserFeedbackComplete` or something like that.

Reply:
I've changed it to `handleUserFeedback`